src/components/TextInput/index.js

```diff
@@ -113,7 +114,10 @@ class TextInput extends Element {
     }
 
     _onInputKeyDown(evt) {
-        if ((evt.keyCode === 27 && this.blurOnEscape) || (evt.keyCode === 13 && this.blurOnEnter)) {
+        if (evt.keyCode === 27 && this.blurOnEscape) {
+            this.value = this._prevValue;
```

Comment:
Won't setting 'value' result in a value recorded when the value has not changed yet? Maybe the value should be changed on the input field directly instead with no 'change' events fired?

Reply:
Yep good shout, i'll switch it over.